packages/core/src/components/section/_section.scss

```diff
@@ -59,6 +59,12 @@ $section-card-padding-compact: $pt-grid-size * 1.5 !default;
       margin-left: auto;
     }
 
+    &-collapse-caret {
```

Comment:
fixes small position change to the caret that was introduced in https://github.com/palantir/blueprint/pull/7144